app/controllers/tasks_controller.rb

```diff
@@ -32,18 +32,24 @@ def update
   end
 
   def show
-    @task = @task_list.find(params[:id])
+    @task = @task_list.tasks.find(params[:id])
```

Comment:
A possible candidate to extract into a method and to be called in a `before_action` block! :)